subscriptions/tasks.py

```diff
@@ -1,16 +1,194 @@
+import requests
+import json
+
 from celery.task import Task
 from celery.utils.log import get_task_logger
 from celery.exceptions import SoftTimeLimitExceeded
 
 from django.conf import settings
 from django.core.exceptions import ObjectDoesNotExist
+from django.db.models import Max
 
 from .models import Subscription
+from contentstore.models import Message
 from scheduler.client import SchedulerApiClient
 
 logger = get_task_logger(__name__)
 
 
+class Send_Next_Message(Task):
+
+    """
+    Task to load and contruct message and send them off
+    """
+    name = "subscriptions.tasks.send_next_message"
+
+    class FailedEventRequest(Exception):
+
+        """
+        The attempted task failed because of a non-200 HTTP return
+        code.
+        """
+
+    def get_identity(self, identity_uuid):
+        url = "%s/%s/%s/" % (settings.IDENTITY_STORE_URL,
+                             "identities", identity_uuid)
+        headers = {'Authorization': ['Token %s' % (
+            settings.IDENTITY_STORE_TOKEN, )],
+            'Content-Type': ['application/json']}
+        r = requests.get(url, headers=headers)
+        return r.json()
+
+    def get_identity_address(self, identity_uuid):
+        url = "%s/%s/%s/addresses/msisdn" % (settings.IDENTITY_STORE_URL,
+                                             "identities", identity_uuid)
+        params = {"default": True}
+        headers = {'Authorization': ['Token %s' % (
+            settings.IDENTITY_STORE_TOKEN, )],
+            'Content-Type': ['application/json']}
+        r = requests.get(url, params=params, headers=headers).json()
+        if len(r["results"]) > 0:
+            return r["results"][0]
+        else:
+            return None
+
+    def run(self, subscription_id, **kwargs):
+        """
+        Load and contruct message and send them off
+        """
+        l = self.get_logger(**kwargs)
+
+        l.info("Loading Subscription")
+        try:
+            subscription = Subscription.objects.get(id=subscription_id)
+            # start here
+            l.info("Loading Message")
+            message = Message.objects.get(
+                messageset=subscription.messageset,
+                sequence_number=subscription.next_sequence_number,
+                lang=subscription.lang)
+            l.info("Loading Initial Recipient Identity")
+            initial_id = self.get_identity(subscription.identity)
+            if "communicate_through" in initial_id and \
+                    ["communicate_through"] is not None:
+                # we should not send messages to this ID. Load the listed one.
+                to_addr = self.get_identity_address(
+                    initial_id["communicate_through"])
+            else:
+                # set recipient data
+                to_addr = self.get_identity_address(subscription.identity)
```

Comment:
You're making an extra Api call, but inter-service calls should be fast enough, so should be okay.